packages/protocol/contracts/common/Accounts.sol

```diff
@@ -396,15 +590,42 @@ contract Accounts is
     }
   }
 
+  /**
+   * @notice Returns the signer for the specified account and role.
+   * @param _account The address of the account.
+   * @param role The role of the signer.
+   */
+  function getDefaultSigner(address _account, string memory role) public view returns (address) {
+    require(isAccount(_account), "Unknown account");
+
+    Account storage account = accounts[_account];
+
+    address signer;
+    if (keccak256(abi.encodePacked(role)) == keccak256(abi.encodePacked(ValidatorSigner))) {
```

Comment:
there is a lot of redundancy in these if/elses (switch)
can we abstract?

Reply:
TBH I don't see how. We have this switch block in four places but I'm not sure what an extracted function would look like as we're doing different things each time.